### PR TITLE
Lolodi/using pipes

### PR DESCRIPTION
**Pull request by lolodi**
Switching back to using pipes.
This should get rid of the XHR errors and make it easier to re-implement proper progress reporting going forward.

### CODE REVIEW

File: src/FileDownloader.ts [hunk 1]
```diff
@@ -58,25 +58,50 @@ export default class FileDownloader implements IFileDownloader {
...
+                onDownloadProgressChange
+            );
+
+            const writeStream: Writable = settings?.shouldUnzip ?? false
+                ? ZipDecompressor(tempFileDownloadPath)
+                : fs.createWriteStream(tempFileDownloadPath);
+
+            // Start the download and wait for it to completely finish writing to disk
+            // The first element in the pipeline must be the readableStream and the last element must be the writableStream
+            //await pipelineAsync([downloadStream, writeStream]);
```
Comment (pragyamehta) [inline src/FileDownloader.ts:90]: Have we kept this on purpose?

File: src/networking/HttpRequestHandler.ts [hunk 2]
```diff
@@ -89,19 +90,22 @@ export default class HttpRequestHandler implements IHttpRequestHandler {
             });
         }
 
-        if (onDownloadProgressChange != null) {
-            const headers: { [key: string]: any } = response.headers;
-            const contentLength = parseInt(headers[`content-length`], 10);
-            const totalBytes = contentLength as number ?? undefined;
-            let downloadedBytes = 0;
+        // We should not feed of the data handler here if we are goign to pipe of the downloadStream later on. this forks the pipe and creates problem e.g. FILE_ENDED, PREMATURE_CLOSE
```
Comment (pragyamehta) [inline src/networking/HttpRequestHandler.ts:93]: nit: spelling

File: src/FileDownloader.ts (hunk 1 above)
Comment (lolodi) [reply src/FileDownloader.ts]: good point, removed.

**Comment by daniv-msft** [comment]
Could you please rebase this PR on the latest commit in this repo? It seems that multiple changes were actually taken care of in the previous PR, and seeing only the latest ones would help reviewing.

**Comment by lolodi** [comment]
> Could you please rebase this PR on the latest commit in this repo? It seems that multiple changes were actually taken care of in the previous PR, and seeing only the latest ones would help reviewing.

Fixed that :)

**Review by daniv-msft — APPROVED** [review]
Looks good to me!

**Pull request merged**